submissions/TaniaBondarenko/frogger_game/app.js

```diff
@@ -0,0 +1,117 @@
+const borders = {
+    rightBorder: 400,
+    leftBorder: 5,
+    upBorder: 410,
+    downBorder: 70
+};
+
+const Enemy = function (x, y) {
+    this.sprite = 'images/enemy-bug.png';
+    this.x = x;
+    this.y = y;
+    this.speed = 75;
+
+    this.render = function () {
+        ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+    };
+};
+
+    Enemy.prototype.update = function (dt) {
+        this.x += this.speed * dt;
+        if (this.x >= 500) {
+            this.x = -10;
+            this.speed = Math.floor(40 + Math.random() * 100);
+        }
+    };
+
+this.checkCollisions = function () {
+    allEnemies.forEach(function (enemy) {
+        let diffX = Math.abs(enemy.x - player.x);
+        let diffY = Math.abs(enemy.y - player.y);
+        let distance = Math.sqrt(Math.pow(diffX, 2) + Math.pow(diffY, 2));
+        if (distance <= 50) {
+            alert(`Oops... \uD83D\uDE3F Try Again`);
+            player.initialCoordinates();
+        } else if (player.y === borders.downBorder) {
+            player.y = 60;
+            setTimeout(function () { alert(`\uD83D\uDE38 Congratulations! You won! \uD83D\uDE38 `); }, 100);
+            setTimeout(function () { player.initialCoordinates(); }, 50);
+        };
+    });
+};
+
+const Player = function (x, y) {
+    this.sprite = "images/char-cat-girl.png";
+    this.x = x;
+    this.y = y;
+    this.stepX = 98;
+    this.stepY = 85;
+    this.initialCoordinates = function () {
```

Comment:
use Player prototype for 'initialCoordinates' definition:
```
Player.prototype.initialCoordinates = function () { 
```

Reply:
okay